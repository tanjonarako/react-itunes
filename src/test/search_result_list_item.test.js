import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchResultListItem from '../components/search_result_list_item';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchResultListItem', () => {
  let item, wrapper;
  beforeEach(() => {
    item = {trackId: 1, trackName: "Shape of you", artistName: "Ed Sheeran", artworkUrl60: "http://is1.mzstatic.com/image/thumb/Music122/v4/ef/ca/a4/efcaa44a-517a-98e3-2353-5bff15471ed5/source/60x60bb.jpg"};
    wrapper = shallow(<SearchResultListItem key={item.trackId} item={item}/>);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an item with props', () => {
    expect(wrapper.find('img').prop('src')).toEqual('http://is1.mzstatic.com/image/thumb/Music122/v4/ef/ca/a4/efcaa44a-517a-98e3-2353-5bff15471ed5/source/60x60bb.jpg');
    expect(wrapper.find('h2').text()).toEqual('Shape of you');
    expect(wrapper.find('span').text()).toEqual('Ed Sheeran');
  });
});
