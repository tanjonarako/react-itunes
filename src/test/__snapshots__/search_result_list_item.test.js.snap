// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultListItem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultListItem
    item={
        Object {
            "artistName": "Ed Sheeran",
            "artworkUrl60": "http://is1.mzstatic.com/image/thumb/Music122/v4/ef/ca/a4/efcaa44a-517a-98e3-2353-5bff15471ed5/source/60x60bb.jpg",
            "trackId": 1,
            "trackName": "Shape of you",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="song logo"
          height="60"
          src="http://is1.mzstatic.com/image/thumb/Music122/v4/ef/ca/a4/efcaa44a-517a-98e3-2353-5bff15471ed5/source/60x60bb.jpg"
          width="60"
/>,
        <div
          className="search-result-list__song-details pl-3 col-flex"
>
          <h2
                    className="search-result-list__song-name m-0"
          >
                    Shape of you
          </h2>
          <span
                    className="search-result-list__song-artist m-0"
          >
                    Ed Sheeran
          </span>
</div>,
      ],
      "className": "search-result-list__song row-flex p-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "song logo",
          "height": "60",
          "src": "http://is1.mzstatic.com/image/thumb/Music122/v4/ef/ca/a4/efcaa44a-517a-98e3-2353-5bff15471ed5/source/60x60bb.jpg",
          "width": "60",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="search-result-list__song-name m-0"
>
              Shape of you
</h2>,
            <span
              className="search-result-list__song-artist m-0"
>
              Ed Sheeran
</span>,
          ],
          "className": "search-result-list__song-details pl-3 col-flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Shape of you",
              "className": "search-result-list__song-name m-0",
            },
            "ref": null,
            "rendered": "Shape of you",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ed Sheeran",
              "className": "search-result-list__song-artist m-0",
            },
            "ref": null,
            "rendered": "Ed Sheeran",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="song logo"
            height="60"
            src="http://is1.mzstatic.com/image/thumb/Music122/v4/ef/ca/a4/efcaa44a-517a-98e3-2353-5bff15471ed5/source/60x60bb.jpg"
            width="60"
/>,
          <div
            className="search-result-list__song-details pl-3 col-flex"
>
            <h2
                        className="search-result-list__song-name m-0"
            >
                        Shape of you
            </h2>
            <span
                        className="search-result-list__song-artist m-0"
            >
                        Ed Sheeran
            </span>
</div>,
        ],
        "className": "search-result-list__song row-flex p-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "song logo",
            "height": "60",
            "src": "http://is1.mzstatic.com/image/thumb/Music122/v4/ef/ca/a4/efcaa44a-517a-98e3-2353-5bff15471ed5/source/60x60bb.jpg",
            "width": "60",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="search-result-list__song-name m-0"
>
                Shape of you
</h2>,
              <span
                className="search-result-list__song-artist m-0"
>
                Ed Sheeran
</span>,
            ],
            "className": "search-result-list__song-details pl-3 col-flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Shape of you",
                "className": "search-result-list__song-name m-0",
              },
              "ref": null,
              "rendered": "Shape of you",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ed Sheeran",
                "className": "search-result-list__song-artist m-0",
              },
              "ref": null,
              "rendered": "Ed Sheeran",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
